controller/server/view/status_page.ml

```diff
@@ -1,24 +1,131 @@
+
+let opt_elem opt = Option.value ~default:[] @@ Option.map (fun e -> [e]) opt
+
+let action_form ?confirm_msg action button_label =
+  form
+    ~a:[ a_action action
+    ; a_method `Post
+    ; a_class [ "d-Status__ActionForm" ]
+    ]
+    [ input
+        ~a:([ a_input_type `Submit
+        ; a_class [ "d-Button" ]
+        ; a_value button_label
+        ] @ (opt_elem (Option.map (fun (m) ->
+                a_onclick (Format.sprintf "return confirm('%s');" m))
+            confirm_msg)))
+        ()
+    ]
+
+let note body = div
+  ~a:[ a_class [ "d-Note" ] ]
+  [ txt body  ]
+
+let reboot_call =
+      [ note "A new version of PlayOS has been installed, reboot to switch to the new version."
+      ; action_form "/system/reboot" "Reboot into updated version"
+      ]
+
+let switch_to_newer_system_call target_slot =
+      [ note "This machine has an out of date PlayOS version selected as the
+              default. You can switch to the new version (requires a reboot)."
+      ; action_form
+            ("/system/switch/" ^ (slot_fmt target_slot))
+            "Switch to newer version and reboot"
+      ]
+
+let switch_to_older_system_call target_slot =
+      [ note "You are running the latest version of PlayOS, but you can still
+              switch back to the older version (requires a reboot)."
+      ; action_form
+            ("/system/switch/" ^ (slot_fmt target_slot))
+            "Switch to older version and reboot"
+      ]
+
+let reinstall_call target_slot  =
+      [ note "The PlayOS installation appears to be faulty, manual system
+              reinstallation is recommended. Please contact support. You can
+              attempt to switch to another system slot (requires reboot)."
+      ; action_form
+            ("/system/switch/" ^ (slot_fmt target_slot))
+            "Switch to other slot and reboot"
+      ]
+
+let factory_reset_call =
+    let confirm_msg = "This will wipe all configuration and login data. Proceed?"
+    in
+      [ note "WARNING: Clears all user data and reboots the machine, resulting in a
+              fresh install state. Will require to manually reconfigure network,
+              localization and all other settings. Any active sessions and/or
+              logins will be expired."
+      ; action_form ~confirm_msg ("/system/factory-reset") "⚠ Factory Reset"
+      ]
+
+let other_slot = let open Rauc.Slot in function
+    | SystemA -> SystemB
+    | SystemB -> SystemA
+
+let suggested_action_of_state (update:Update.state) (rauc:rauc_state) booted_slot =
+    let target_slot = other_slot booted_slot in
+    match (update, rauc) with
+        | (RebootRequired, _) ->
+            Some (Definition.description reboot_call)
+        | (OutOfDateVersionSelected, Status _) ->
+            Some (Definition.description (
+                switch_to_newer_system_call target_slot
+            ))
+        | (UpToDate {booted; inactive}, Status _) when booted != inactive ->
```

Comment:
```suggestion
        | (UpToDate {booted; inactive}, Status _) when booted <> inactive ->
```

Oops, missed this before, need to use [structural equality op](https://ocaml.org/manual/5.2/expr.html#ss%3Aexpr-operators).